src/client.js

```diff
@@ -218,7 +218,7 @@ class Client extends EventEmitter {
     if (!this.serializer.writable) { return }
     debug('writing packet ' + this.state + '.' + name)
     debug(params)
-    this.serializer.write({ name, params })
+    this.serializer.write({ name, params }, (e) => { if (e) throw e })
```

Comment:
This doesn't happen here
It's the error event which is already handled

Reply:
This is the callback that is propagated from errors in _transform. The write method is here:
https://github.com/nodejs/readable-stream/blob/master/lib/_stream_writable.js#L288-L310

Reply:
If you don't pass a callback the errors in transform get dropped

Reply:
ok then we should emit an error with the err event in this case
crashing the program is not the best imo
users could have a larger program on top of this, that could for example decide to restart the client with another version/server